code/modules/admin/verbs/adminpm.dm

```diff
@@ -185,3 +185,192 @@
 			continue
 		if(X.holder.rights & R_ADMIN|R_MOD)
 			to_chat(X, "<span class='pm'><span class='other'>" + create_text_tag("pm_other", "PM:", X) + " <span class='name'>[key_name(src, X, 0)]</span> to <span class='name'>[sender]</span>: <span class='message'>[msg]</span></span></span>")
+*/
+
+//allows right clicking mobs to send an admin PM to their client, forwards the selected mob's client to cmd_admin_pm
+/client/proc/cmd_admin_pm_context(mob/M as mob in SSmobs.mob_list)
```

Comment:
as long as you're replacing this, delete the old commented out version

Reply:
You mean the old adminpm system?

Reply:
I'd prefer to keep them, until the system's been battletested, so it's easy to revert.

Reply:
reverting won't be done by manually editing code, that's what we have git for